app/components/widgets/CustomFieldsEditor.vue.ts

```diff
@@ -0,0 +1,224 @@
+import Vue from 'vue';
+import { cloneDeep } from 'lodash';
+import { Component, Prop, Watch } from 'vue-property-decorator';
+import { codemirror } from 'vue-codemirror';
+import { CodeInput } from 'components/shared/inputs/inputs';
+import { IWidgetData, WidgetSettingsService } from 'services/widget-settings/widget-settings';
+import { Inject } from '../../util/injector';
+import { WidgetsService } from 'services/widgets';
+import { $t } from 'services/i18n/index';
+import { IInputMetadata, metadata } from 'components/shared/inputs';
+import FormGroup from 'components/shared/inputs/FormGroup.vue';
+import { debounce } from 'lodash-decorators';
+
+type TCustomFieldType =
+  'colorpicker' |
+  'slider' |
+  'textfield' |
+  'fontpicker' |
+  'dropdown' |
+  'image-input' |
+  'sound-input';
+
+export interface ICustomField {
+  label: string;
+  type: TCustomFieldType;
+  value: string | number;
+
+  options?: Dictionary<string>;
+  max?: number;
+  min?: number;
+  steps?: number;
+}
+
+const DEFAULT_CUSTOM_FIELDS: Dictionary<ICustomField> = {
+
+  customField1: {
+    label: 'Color Picker Example',
+    type: 'colorpicker',
+    value: '#000EF0'
+  },
+
+  customField2: {
+    label: 'Slider Example',
+    type: 'slider',
+    value: 3,
+    max: 200,
+    min: 100,
+    steps: 4
+  },
+
+  customField3: {
+    label: 'Textfield Example',
+    type: 'textfield',
+    value: 'Hi There'
+  },
+
+  customField4: {
+    label: 'Font Picker Example',
+    type: 'fontpicker',
+    value: 'Open Sans'
+  },
+
+  customField5: {
+    label: 'Dropdown Example',
+    type: 'dropdown',
+    options: {
+      optionA: 'Option A',
+      optionB: 'Option B',
+      optionC: 'Option C'
+    },
+    value: 'optionB'
+  },
+
+  // TODO:
```

Comment:
Is there are reason these are commented out? SoundInput is in and and think `image-input` is the just the MediaGalleryInput

Reply:
does it have an option to allow to select only images? Didn't see this option

Reply:
nope! but afaik neither does the site have that. now that `sound-input` is in place though basically all the architecture is there to add that validation. if that's blocking i can do it now, otherwise i think we would probably be ok just using `media-gallery-input`

Reply:
Yes, let's add a filter option for media gallery to show only images. It's not a blocker since we can add it here in the separated PR